### PR TITLE
chore: Adds support for MacT capture using HTTP_MOCKER_CAPTURE

### PR DESCRIPTION
**Pull request by EspenAlbert**
### In scope
- All files in the `internal/testutil` directory
- Any comments on the provider mocking, naming of env-vars, call experience, etc.

### Outside scope
- All other files
- Detailed review of yaml dumping and how-it-works, this will be better explained in Outcomes document later

### Features
- Adds support for MacT capture using HTTP_MOCKER_CAPTURE and replay using HTTP_MOCKER_REPLAY
- Refactor advanced cluster validation tests and remove obsolete unit tests (refactored/added in next PR #2875 )

See example of generated files for many of the current SDKv2 implementation tests in:
https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2885/commits/09847fcc338c8a1b1a5ab1fc4e5570eb1e0d70d6

Link to any related issue(s): #2875 and #2885

PS might be reverted based on https://jira.mongodb.org/browse/CLOUDP-287330

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/testutil/unit/main_test.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"os"
+	"testing"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/unit"
+)
+
+func TestMain(m *testing.M) {
+	unit.InitializeAPISpecPaths()
```
Comment (lantoli) [inline internal/testutil/unit/main_test.go:11]: is InitializeAPISpecPaths being called twice, from TestMain and init?

File: internal/testutil/unit/http_mocker_api_paths.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+type OpenapiSchema struct {
+	Paths map[string]map[string]interface{} `yaml:"paths"`
```
Comment (lantoli) [inline internal/testutil/unit/http_mocker_api_paths.go:17]: there are 3 interface{} than be changed to any

File: internal/testutil/unit/http_mocker_data.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+	}
+	return &yaml.Node{
+		Kind:    yaml.MappingNode,
+		Style:   yaml.FoldedStyle,
+		Content: childNodes,
+	}, nil
+}
+
+func (i *RequestInfo) id() string {
+	return fmt.Sprintf("%s_%s_%s_%s", i.Method, i.Path, i.Version, i.Text)
```
Comment (lantoli) [inline internal/testutil/unit/http_mocker_data.go]: nit: could use idShort logic and add text

**Review by lantoli — APPROVED** [review]

File: internal/testutil/unit/main_test.go (hunk 1 above)
Comment (EspenAlbert) [reply internal/testutil/unit/main_test.go:11]: Removed in 41e4fb989e98a4f4f5f5da90d296d3195647d204

File: internal/testutil/unit/http_mocker_api_paths.go (hunk 2 above)
Comment (EspenAlbert) [reply internal/testutil/unit/http_mocker_api_paths.go:17]: changed in 41e4fb989e98a4f4f5f5da90d296d3195647d204

File: internal/testutil/unit/http_mocker_data.go (hunk 3 above)
Comment (EspenAlbert) [reply internal/testutil/unit/http_mocker_data.go]: nice. Changed in 41e4fb989e98a4f4f5f5da90d296d3195647d204

File: internal/service/advancedclustertpf/resource_schema_test.go [hunk 4]
```diff
@@ -10,6 +10,10 @@ import (
 )
 
 func TestAdvancedCluster_ValidationErrors(t *testing.T) {
+	const (
+		projectID   = "111111111111111111111111"
```
Comment (lantoli) [inline internal/service/advancedclustertpf/resource_schema_test.go:14]: why wasn't this test failing before, do we also have projectID and clusterName at package level? same for next test

File: internal/testutil/unit/http_mocker_api_paths.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+		gitDir := path.Join(parentCandidate...)
+		if fileExist(gitDir) {
+			repoPath, _ := strings.CutSuffix(gitDir, ".git")
+			return path.Join(repoPath, relPath)
+		}
+	}
+	panic("could not find repo root")
+}
+
+func init() {
```
Comment (lantoli) [inline internal/testutil/unit/http_mocker_api_paths.go:117]: it's normally better to restrict use of init, but i think it's ok here

**Review by lantoli — APPROVED** [review]
LGTM

File: internal/service/advancedclustertpf/resource_schema_test.go (hunk 4 above)
Comment (EspenAlbert) [reply internal/service/advancedclustertpf/resource_schema_test.go:14]: Removed these from the package level to avoid accidental usage

**Comment by EspenAlbert** [comment]
Tests passing in https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/12328171153/job/34411139681 merging this

**Pull request merged**